### PR TITLE
Cache name normalization

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
Profiling Ankidroid, I realized that ankidroid spent 12 seconds on
startup in the method normalize. That's far too much. I actually have
trouble believing the profiler here. (More precisely it mostly spent
time in "toLowerCase")

Anyway, I added this simple caching method, so that it becomes far
quicker. After all, caching is not context dependant, it always give
the same output to the same input, so we never have any need to empty
the cache, so that's perfect here

## Fixes
Anki startup still takes more than half a minute

## Approach
Profiling and checking where time is spent

## How Has This Been Tested?
Installing on the phone and using it. Looking at profiler and realizing that the time decreased in this method.

## Learning (optional, can help others)
Profilers are hard to use

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java [hunk 1]
```diff
@@ -1003,10 +1003,12 @@ public boolean isDyn(long did) {
      * utils methods
      * **************************************
      */
+    private static HashMap<String, String> normalized = new HashMap<String, String>();
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java:1006]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226&bid=17366027)

**Comment by Arthur-Milchior** [comment]
Do you want me to move the field in top of the class ? I don't think it would help the code to be more readable in this case. But I can do it if you want.

**Comment by mikehardy** [comment]
no - one of those cases where codacy rule is good in general but common sense and current position is better. i'll merge as is

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226&bid=17366027)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java [hunk 2]
```diff
@@ -1007,10 +1007,12 @@ public boolean isDyn(long did) {
...
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java:1010]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226&bid=17366027)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java (hunk 2 above)
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java:1010]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5306226)

**Comment by mikehardy** [comment]
This released as 2.10alpha65 as soon as google is done processing it

**Pull request merged**